### ISSUE TITLE
Registrering/innlogging med **Strava OAuth**

### ISSUE THREAD

**Issue by limekex** [limekex/tvs-infra#29]
## Summary
Registrering/innlogging med **Strava OAuth** som oppretter (eller kobler til) en WordPress-bruker med rollen **tvs_athlete**. Denne rollen har **kun frontend-tilgang** (ingen wp-admin UI). Etter suksessfull innlogging settes auth-cookie, tokens lagres i user_meta, og bruker redirectes til en frontend «konto»-side.

## Scope
- Ny WP-rolle **tvs_athlete** på plugin-aktivisering.
- Strava OAuth «register/login» flyt:
  - Hvis Strava athlete-id finnes i `user_meta` → logg inn den brukeren.
  - Hvis ikke → opprett ny bruker (rolle: `tvs_athlete`), lagre tokens + athlete-id, logg inn.
- **Ingen** backend-UI for `tvs_athlete`:
  - Skjul admin bar for rollen.
  - Hindre tilgang til `wp-admin` (redirect til frontend), unntak for `admin-ajax.php` og REST.
- Frontend-sider:
  - `/login` med «Logg inn med Strava»-knapp.
  - `/account` (frontend) som viser basisprofil og status (f.eks. gratis-kvote).
- Tekst-domene: `tvs-virtual-sports`.

## Acceptance Criteria (AC)
- [ ] Bruker kan trykke «Logg inn med Strava», fullføre consent, lande på site innlogget som `tvs_athlete`.
- [ ] Eksisterende Strava-bruker blir **ikke** duplisert; login gjenkjenner athlete-id.
- [ ] `tvs_athlete` ser **ikke** admin bar og blir redirectet bort fra `wp-admin/*`.
- [ ] Tokens (`access`, `refresh`, `expires_at`) lagres i `user_meta` og oppdateres ved refresh.
- [ ] Frontend `/account` viser innlogget tilstand.
- [ ] Compliance til Strava Guidelines https://developers.strava.com/guidelines/

## Implementation notes
**Backend**
- Files:
  - `wp-content/plugins/tvs-virtual-sports/includes/class-tvs-auth.php` (NY)
  - `wp-content/plugins/tvs-virtual-sports/includes/class-tvs-roles.php` (NY)
  - `wp-content/plugins/tvs-virtual-sports/includes/class-tvs-rest.php` (utvid)
  - `wp-content/plugins/tvs-virtual-sports/includes/class-tvs-strava.php` (gjenbruk)
  - `wp-content/plugins/tvs-virtual-sports/tvs-virtual-sports.php` (hooks)
- Rolleoppsett:
  ```php
  register_activation_hook( __FILE__, function(){
    add_role('tvs_athlete','Athlete',[
      'read' => true, // ingen andre caps
    ]);
  });
  ```
- Admin-tilgang/toolbar:
  ```php
  add_filter('show_admin_bar', function($show){
    if ( current_user_can('tvs_athlete') ) return false;
    return $show;
  });
  add_action('admin_init', function(){
    if ( is_admin() && !wp_doing_ajax() && current_user_can('tvs_athlete') ) {
      wp_safe_redirect( home_url('/account') ); exit;
    }
  });
  ```
- OAuth-flyt (gjenbruker `POST /tvs/v1/strava/connect`):
  - Etter `exchange_code_for_tokens($code)`, hent athlete-info (id, navn).
  - Søk bruker via `get_users(['meta_key'=>'tvs_strava_athlete_id','meta_value'=>$athlete_id])`.
  - Hvis ingen:
    ```php
    $uid = wp_insert_user([
      'user_login' => 'strava_'.$athlete_id,
      'user_pass'  => wp_generate_password(24),
      'user_email' => 'athlete+' . $athlete_id . '@example.invalid',
      'role'       => 'tvs_athlete',
      'display_name' => $athlete_name ?: 'Strava Athlete'
    ]);
    update_user_meta($uid,'tvs_strava_athlete_id',$athlete_id);
    ```
  - Lagre tokens i `user_meta('tvs_strava_tokens')`.
  - Logg inn:
    ```php
    wp_set_current_user($uid);
    wp_set_auth_cookie($uid,true);
    ```
  - Returner 200 med `{ logged_in: true, user: { id, role } }`.
- **Scopes** anbefalt:
  - Minimum for innlogging + opplasting: `read,activity:write`.
  - For rute-import senere: `read_all,profile:read_all` (kan be om i eget trinn).
- I18n:
  - Alle strenger som vises (`__()`, `_x()`, `esc_html__()`) med text-domain `tvs-virtual-sports`.

**Frontend**
- `/login`:
  - Knapp «Logg inn med Strava» → `https://www.strava.com/oauth/authorize?...&client_id=...&redirect_uri={site}/strava/connected&response_type=code&scope=read,activity:write`.
- `/strava/connected`:
  - JS som POSTer `code` til `/wp-json/tvs/v1/strava/connect` (med `X-WP-Nonce`), deretter `location.href='/account?welcome=1'`.
- `/account`:
  - Vis innlogget bruker, «Koble fra Strava» (senere), og kvote (fra gating-issue).

**Sikkerhet & personvern**
- Nonce/CSRF på POST.
- Tokens i user_meta – ikke eksponer dem i REST.
- «Koble fra Strava» (senere): slette tokens/meta.

## Testplan
**Manuell**
1. Gå til `/login` → «Logg inn med Strava» → consent → redirect til `/strava/connected?code=...`.
2. Bekreft at bruker blir opprettet/innlogget og landet på `/account`.
3. Forsøk å gå til `/wp-admin/` → redirect til `/account` og ingen adminbar.
**Automatisert**
- PHPUnit: 
  - Opprett/innlogg med mock av `exchange_code_for_tokens`.
  - Rolle/redirect tester.
  - At `user_meta('tvs_strava_athlete_id')` settes én gang (ingen duplikat).

## Release/Docs
- Added: Ny rolle `tvs_athlete`, Strava-login.
- Docs: kort README-del «Registrering/innlogging med Strava».

## Meta
Labels: `type:feature`, `area:backend`, `area:frontend`, `prio:P1`
Milestone: `v1.2.0`
